### PR TITLE
json: a comprehensive fix for oss-fuzz#10908

### PR DESCRIPTION
**Pull request by poettering**
Mostly minor fixes plus: 

1. Two fixes for the stack overflow by recursion found by oss-fuzz#10908 (one that fixes the specific issue itself, and another one that makes the whole class of issues go away)
2. An optimization so that we can process static const literal strings directly as JsonVariant* objects without any additional allocation

### CODE REVIEW

**Comment by evverx** [comment]
@poettering could you put the reproducer from https://oss-fuzz.com/download?testcase_id=5639441482252288 to `test/fuzz/fuzz-json`?

**Comment by poettering** [comment]
@evverx ah, i actually did, but forgot to "git add" it ;-)

Fixed now. PTAL.

File: src/test/test-json.c [hunk 1]
```diff
@@ -407,5 +439,7 @@ int main(int argc, char *argv[]) {
 
         test_source();
 
+        test_depth();
```
Comment (keszybz) [inline src/test/test-json.c:443]: Comes with tests, nice!

File: src/basic/json.c [hunk 2]
```diff
@@ -589,6 +589,10 @@ int json_variant_new_object(JsonVariant **ret, JsonVariant **array, size_t n) {
                         *c = array[v->n_elements];
                 uint16_t d;
 
+                if ((v->n_elements & 1) == 0 &&
```
Comment (keszybz) [inline src/basic/json.c:623]: I'd spell this as `v->n_elements % 2 == 0`, but this works too.

**Pull request merged**